scarb/src/compiler/db.rs

```diff
@@ -39,10 +42,19 @@ fn build_project_config(unit: &CompilationUnit) -> Result<ProjectConfig> {
         .iter()
         .filter(|component| !component.package.id.is_core())
         .map(|component| {
-            (
-                component.cairo_package_name(),
-                component.target.source_root().into(),
-            )
+            let crate_root: CrateRootPath = if component
+                .target
+                .source_path
+                .clone()
+                .ends_with(DEFAULT_SOURCE_PATH)
+            {
+                CrateRootPath::Simple(component.target.source_root().into())
+            } else {
+                CrateRootPath::Entrypoint(DetailedEntrypointPath {
+                    entrypoint: component.target.source_path.clone().into(),
+                })
+            };
+            (component.cairo_package_name(), crate_root)
```

Comment:
ugh, this doesn't sound good as I see it now. This seems to be easily unifyable on API surface, perhaps we should fix compiler internal issues blocking this instead of convoluting its API?